### PR TITLE
Update lint

### PR DESCRIPTION
**Pull request by hildjj**
Updated all deps (including in `@peggyjs/eslint-config`), and fixed a few minor lint nits.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -22,15 +22,6 @@ module.exports = {
         sourceType: "module",
         ecmaVersion: 2018,
       },
-      rules: {
```
Comment (hildjj) [inline .eslintrc.js:25]: moved to `@peggyjs/eslint-config`

File: lib/peg.d.ts [hunk 2]
```diff
@@ -119,7 +119,7 @@ declare namespace ast {
      */
     nameLocation: LocationRange;
     /** Parsing expression of this rule. */
-    expression: Named | Expression;
+    expression: Expression | Named;
```
Comment (hildjj) [inline lib/peg.d.ts:122]: Most of these are just sorting issues.

File: lib/peg.d.ts [hunk 3]
```diff
@@ -308,21 +304,21 @@ type Problem = [
 /** Thrown if the grammar contains a semantic error. */
 export class GrammarError extends Error {
   /** Location of the error in the source. */
-  location?: LocationRange;
+  public location?: LocationRange;
```
Comment (hildjj) [inline lib/peg.d.ts:307]: Added a lint rule that said that all fields and methods needed explicit visibility, which I like.  Kept these public to not have a breaking change.

File: lib/peg.d.ts [hunk 4]
```diff
@@ -846,23 +844,22 @@ export interface ParserOptions {
...
 export interface ParserTracer {
   trace(event: ParserTracerEvent): void;
 }
 
-export type ParserTracerEvent =
-  | { type: "rule.enter"; rule: string; location: LocationRange }
-  | { type: "rule.match"; rule: string; result: any; location: LocationRange }
-  | { type: "rule.fail"; rule: string; location: LocationRange };
+export type ParserTracerEvent
+  = { type: "rule.enter"; rule: string; location: LocationRange }
```
Comment (hildjj) [inline lib/peg.d.ts:860]: I was shocked that the sorting rule found these.

File: test/cli/run.spec.ts [hunk 5]
```diff
@@ -89,24 +89,24 @@ ${str}`);
   }
 }
 
-type Options = {
+interface Options {
   args?: string[];
   encoding?: BufferEncoding;
-  env?: Record<string, string>;
```
Comment (hildjj) [inline test/cli/run.spec.ts:95]: I didn't have  strong preference for `Record` vs `{[index]}`, and we use the latter more often.

File: test/types/peg.test-d.ts [hunk 6]
```diff
@@ -1,7 +1,6 @@
 import * as peggy from "../..";
 import tsd, { expectType } from "tsd";
-import { SourceNode } from "source-map-generator";
-import formatter from "tsd/dist/lib/formatter";
```
Comment (hildjj) [inline test/types/peg.test-d.ts:4]: formatter is no longer exposed, and this is just for our own ease of use.

**Review by Mingun — APPROVED** [review]

**Pull request merged**